### PR TITLE
Implement WlOutput and XdgOutput

### PR DESCRIPTION
**Pull request by PolyMeilex**

### CODE REVIEW

File: src/wayland/output/handlers.rs [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        let output = data_init.init(
+            resource,
+            OutputUserData {
+                global_data: global_data.clone(),
+            },
+        );
+
+        let mut inner = global_data.inner.0.lock().unwrap();
+
+        trace!(inner.log, "New global instantiated.");
```
Comment (vberger) [inline src/wayland/output/handlers.rs:45]: If we're going to have these trace messages, maybe they could be a little more informative, maybe at least tell that it is an output that is instantiated.

File: src/wayland/output/mod.rs [hunk 2]
```diff
@@ -339,82 +355,95 @@ impl Output {
...
     /// with the matching client output
-    pub fn enter(&self, surface: &wl_surface::WlSurface) {
-        if let Some(client) = surface.as_ref().client() {
-            self.with_client_outputs(client, |output| surface.enter(output))
+    pub fn enter(&self, dh: &mut DisplayHandle<'_>, surface: &wl_surface::WlSurface) {
+        let client = dh.get_client(surface.id()).ok();
+
+        if let Some(client) = client {
+            self.with_client_outputs(dh, &client, |dh, output| surface.enter(dh, output))
         }
```
Comment (vberger) [inline src/wayland/output/mod.rs:425]: If we're going to just match it using `if let`, what's the point of using `.ok()` here?

File: src/wayland/output/mod.rs [hunk 3]
```diff
@@ -339,82 +355,95 @@ impl Output {
...
     /// with the matching client output
-    pub fn leave(&self, surface: &wl_surface::WlSurface) {
-        if let Some(client) = surface.as_ref().client() {
-            self.with_client_outputs(client, |output| surface.leave(output))
+    pub fn leave(&self, dh: &mut DisplayHandle<'_>, surface: &wl_surface::WlSurface) {
+        let client = dh.get_client(surface.id()).ok();
+
+        if let Some(client) = client {
+            self.with_client_outputs(dh, &client, |dh, output| surface.leave(dh, output))
         }
```
Comment (vberger) [inline src/wayland/output/mod.rs:435]: Same here

File: src/wayland/output/handlers.rs (hunk 1 above)
Comment (PolyMeilex) [reply src/wayland/output/handlers.rs]: Indeed, not it does say output global, and prints name of the output as well

**Pull request merged**